tests/plugins_t/test_auditor.py

```diff
@@ -0,0 +1,118 @@
+        cls.mock_auditorclientbuilder = cls.mock_auditorclientbuilder_patcher.start()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.mock_config_patcher.stop()
+        cls.mock_auditorclientbuilder_patcher.stop()
+
+    def setUp(self):
+        self.address = "127.0.0.1"
+        self.port = 8000
+        self.timeout = 20
+        self.user = "user-1"
+        self.group = "group-1"
+        self.site = "testsite"
+        self.cores = 12
+        self.memory = 100
+        self.drone_uuid = "test-drone"
+        self.machine_type = "test_machine_type"
+        config = self.mock_config.return_value
+        config.Plugins.Auditor.host = self.address
+        config.Plugins.Auditor.port = self.port
+        config.Plugins.Auditor.timeout = self.timeout
+        config.Plugins.Auditor.user = self.user
+        config.Plugins.Auditor.group = self.group
+        config.Plugins.Auditor.components.test_machine_type = AttributeDict(
+            Cores=AttributeDict(HEPSPEC=1.2, BENCHMARK=3.0),
+            Memory=AttributeDict(BLUBB=1.4),
+        )
+        config.Sites = [AttributeDict(name=self.site)]
+        config.testsite.MachineTypes = [self.machine_type]
+        config.testsite.MachineMetaData = AttributeDict(
+            test_machine_type=AttributeDict(Cores=self.cores, Memory=self.memory)
+        )
+
+        self.test_param = AttributeDict(
+            site_name=self.site,
+            machine_type=self.machine_type,
+            created=datetime.now(),
+            updated=datetime.now(),
+            resource_status=ResourceStatus.Booting,
+            drone_uuid=self.drone_uuid,
+        )
+
+        builder = self.mock_auditorclientbuilder.return_value
+        builder = builder.address.return_value
+        builder = builder.timeout.return_value
+        self.client = builder.build.return_value
+        self.client.add.return_value = async_return()
+        self.client.update.return_value = async_return()
+
+        self.plugin = Auditor()
+
+    def test_notify(self):
+        self.mock_auditorclientbuilder.return_value.address.assert_called_with(
+            self.address,
+            self.port,
+        )
+        self.mock_auditorclientbuilder.return_value.address.return_value.timeout.assert_called_with(
+            self.timeout,
+        )
+        run_async(
+            self.plugin.notify,
+            state=AvailableState(),
+            resource_attributes=self.test_param,
+        )
+        assert self.client.add.call_count == 1
+        assert self.client.update.call_count == 0
+
+        run_async(
+            self.plugin.notify,
+            state=DownState(),
+            resource_attributes=self.test_param,
+        )
+        assert self.client.add.call_count == 1
+        assert self.client.update.call_count == 1
+
+    def test_construct_record(self):
+        record = self.plugin.construct_record(resource_attributes=self.test_param)
+
+        assert record.record_id == self.drone_uuid
```

Comment:
`TARDIS` is using the `unittest` package, so I would prefer to use `self.assertEqual`calls instead of `assert`.

Reply:
Thanks for the feedback, I forgot about that. In future pyauditor versions I'll try to implement `__eq__` for `Record`, `Component` and `Score`, then these checks should be less verbose.